### PR TITLE
[APM-662] Improve ExecutionNodes Memory Tracking

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

Better and more precise memory tracking in ExecutionNodes.
This is required, as those types of nodes are being used during AQL queries and currently not counted into used memory. Which conflicts to the maximum amount of memory a user can define for each single query execution. 

This PR improves improves the memory accounting for simple types (e.g. bools, ptr, etc.). 
Each used Node differs from each other, so they need to be accounted by their own.

This is the first pr of a pr series. Follow-up prs will take this further. 
There are multiple Node classes which require more memory tracking code, as to old an the state in this PR. 
(e.g. all Nodes which inherit from GraphNode)

**See EE PR linked below (!)**

- [ ] :hankey: Bugfix
- [x] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*

#### Related Information

- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/1291
- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/APM-662

### CODE REVIEW

File: arangod/Aql/ClusterNodes.cpp [hunk 1]
```diff
@@ -184,6 +184,10 @@ CostEstimate RemoteNode::estimateCost() const {
   return estimate;
 }
 
+size_t RemoteNode::getMemoryUsedBytes() const {
+  return sizeof(decltype(*this));
```
Comment (MBkkt) [inline arangod/Aql/ClusterNodes.cpp:188]: Just sizeof(*this)

Comment (markuspf) [reply arangod/Aql/ClusterNodes.cpp:188]: Why?

Comment (MBkkt) [reply arangod/Aql/ClusterNodes.cpp:188]: Because it's simpler?

**Review by jsteemann — APPROVED** [review]
LGTM

**Pull request merged**